src/fft_fftw3.f90

```diff
@@ -37,6 +37,29 @@ end subroutine fftw_cleanup
 
    integer, parameter, public :: D2D_FFT_BACKEND = D2D_FFT_BACKEND_FFTW3
 
+   ! Derived type with all the quantities needed to perform FFT
+   type decomp_2d_fft_engine
+      ! Engine-specific stuff
+      type(c_ptr), private :: plan(-1:2, 3)
+      ! All the engines have this
```

Comment:
Would it be better to do extended types? e.g.
```
! In FFT_common?
type decomp_2d_fft_engine
  ! All the engines have this ...
end type

! In fft_fftw3
type, extends(decomp_2d_fft_engine) :: decomp_2d_fftw3_engine
  ! Engine-specific stuff
  type(c_ptr), private :: plan(-1:2, 3)
end type
```
this should cut down on duplication.

Reply:
I am not sure about the exact syntax for extended types. It is not widely used in the code currently. The code in this PR is similar to the code in the library, and code duplication is avoided using `#include`.